### ISSUE TITLE
loadMultiple not loading posted data

### ISSUE THREAD

**Issue by fsnightmckngbrd** [yiisoft/yii2#17654]
I have a problem regarding loadMultiple. I'm sending data through a custom ajax post that I made. The data were not sent from a form. I don't encounter any errors but it seems that the posted data were not loaded onto the model.

This is my script.
```
var data = {};

$('.txtbox').each(function() {
 var id = $(this).data('txt-id');
 data[id] = {
  value: 'test'
 }
});

$.ajax({
 url: url,
 method: post,
 data: data,
 dataType: 'json'
});
```

And here is the php.
```
$post = Yii::$app->request->post();
$model = Sample::find($id)->all();

if (Mode::loadMultiple($model, $post) && Model::validateMultiple($model)) {
 foreach($model as $value) {
  $data[$value->id] = [
   'value' => $value->value
  ];
 }
}

return json_encode($data);
```

I'm expecting that the **value** will have a **test** value but it contains **null** value. No errors encountered.

Please help. Thanks!

**Comment by fsnightmckngbrd** [comment]
I've also checked params section on the browser, I'm also passing the _csrf that I got from a <meta>.
Here is the sample format of the posted data:

```
_csrf: (some random key generated by Yii2 once the page is loaded)
DocumentTemplateValue[68][value]: test
DocumentTemplateValue[69][value]: test
```

And here is the returned data by the php:
```
68: {value: null}
69: {value: null}
```

**Comment by alex-code** [comment]
You need to specify the id as its own data and group with a unique index.
e.g.
```
DocumentTemplateValue[1][id]: 68
DocumentTemplateValue[1][value]: test
//
DocumentTemplateValue[2][id]: 69
DocumentTemplateValue[2][value]: test
```
```js
const data = {};
$('.txtbox').each(idx => {
  let id = $(this).data('txt-id');
  data[idx] = {
    id: id,
    value: 'test'
  }
});
```

**Comment by fsnightmckngbrd** [comment]
That was my previous code before I posted this. I removed the id because I checked the posted data by a dynamic form used on our other website and it doesn't contain the id. Anyways, I'll give you an update later once I get back on the office and I'll try to put the id back again. Thank you.

**Comment by fsnightmckngbrd** [comment]
Okay, I think I figured out the problem.

After putting back my previous code wherein I have included the indexes on the posted data, loadMultiple() still does not working properly. So I tried to change my loop, wherein I have started the index from 0 instead of the index of the data that I need to post.

I guess loadMultiple() does not work properly if the posted data does not start from 0 index. I thought the index of the posted data will be the id of the model when loaded. Problem solved. Thanks!

Posted data : (Does not work)
```
DocumentTemplateValue[68][id]: 68
DocumentTemplateValue[68][value]: test
```

Posted data : (Works)
```
DocumentTemplateValue[0][id]: 68
DocumentTemplateValue[0][value]: test
```
